### ISSUE TITLE
Duplicate reasons are reported for the same rule

### ISSUE THREAD

**Issue by BernieWhite** [microsoft/PSRule#2553]
**Description of the issue**

When collecting reasons for a rule. Duplicate reasons can occur down different code paths. A rule should not report the exact same reason more then once.

**Error output**

```text
[FAIL] Azure.Deployment.AdminUsername (AZR-000284)
| Template: modules/virtual-machine-windows/v0/.tests/main.tests.bicep:0:0

| RECOMMEND:
| Sensitive properties should be passed as parameters. Avoid using deterministic
| values for sensitive properties.

| REASON:
| - The property 'adminUsername' uses a deterministic literal value.
| - The property 'adminUsername' uses a deterministic literal value.
| - The property 'adminUsername' uses a deterministic literal value.
| - The property 'adminUsername' uses a deterministic literal value.
| - The property 'adminUsername' uses a deterministic literal value.
| - The property 'adminUsername' uses a deterministic literal value.
| - The property 'adminUsername' uses a deterministic literal value.
| - The property 'adminUsername' uses a deterministic literal value.

| HELP:
| - https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.Deployment.AdminUsername/
```

**Module in use and version:**

- Module: PSRule
- Version: **v3.0.0-B0267**
